### PR TITLE
mBed online compiler fix + struct example

### PR DESCRIPTION
**Pull request by pilotak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,4 +43,59 @@ int main() {
...
+};
+
+int main() {
+    printf("i2cee struct test\n");
+
+     // No. of bytes to be stored, but topped up to be multiplied by block size
+    unsigned int setting_block_size = ceil(sizeof(setting)/(double)BLOCK_SIZE)*BLOCK_SIZE; 
+
+    // Temporary buffer
+    char buffer[setting_block_size];
```
Comment (geky) [inline README.md]: My only qualm, is that this should probably be dynamically allocated. With BLOCK_SIZE == 32 this is fine, but if we reach a point where we adopt NAND flash, their block sizes can get up into the 4096+ byte range, which runs the risk of stack overflow.

A bit of a nit, but is probably important on the example we show.

File: README.md [hunk 2]
```diff
@@ -41,4 +43,59 @@ int main() {
...
+    memset(buffer, 0, sizeof(buffer)); // empty buffer, not nessesary
+
+    if(i2cee.read(buffer, 0, setting_block_size ) == 0){ // get data into buffer
+        // Convert what we read into struct
+        memcpy(&tmp, buffer, sizeof(tmp)); // copy only size of struct not setting_block_size
+        printf("\nTemporary struct version: %u, name: %s\n", tmp.version, tmp.name); 
+    }
+    else {
+        printf("Error when reading\n");
+    }
```
Comment (geky) [inline README.md]: If you're changing this file, this should have a bit of a style update:
``` diff
+ if(i2cee.read(buffer, 0, setting_block_size ) == 0){ // get data into buffer
- if (i2cee.read(buffer, 0, setting_block_size) == 0) { // get data into buffer
      // Convert what we read into struct
      memcpy(&tmp, buffer, sizeof(tmp)); // copy only size of struct not setting_block_size
      printf("\nTemporary struct version: %u, name: %s\n", tmp.version, tmp.name); 
+ } else {
- }
- else {
      printf("Error when reading\n");
  }
```

**Review by geky — COMMENTED** [review]
This is a good example to have!

Just a bit of feedback, I'm being a bit picky since this is an example of the front of the repo.

File: README.md (hunk 1 above)
Comment (pilotak) [reply README.md]: You are right, it should be. But as this is a eeprom branch of block memory library and size of eeprom needs to be set by user, it should be fine.

File: README.md (hunk 2 above)
Comment (pilotak) [reply README.md]: I'm fine with that 👍

**Pull request merged**